### PR TITLE
More lint pass merges

### PR DESCRIPTION
**Pull request by Jarcho**
changelog: None

### CODE REVIEW

File: clippy_lints/src/methods/range_zip_with_len.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+use rustc_hir::{Expr, ExprKind, QPath};
+use rustc_lint::LateContext;
+use rustc_span::sym;
+
+use super::RANGE_ZIP_WITH_LEN;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>, recv: &'tcx Expr<'_>, zip_arg: &'tcx Expr<'_>) {
+    if_chain! {
+        if let ExprKind::MethodCall(iter_path, [iter_recv], _) = recv.kind;
+        if iter_path.ident.name == sym::iter;
```
Comment (flip1995) [inline clippy_lints/src/methods/range_zip_with_len.rs:15]: NIT: Could this be done in the `methods` module and the `iter_recv` passed into this function?

**Review by flip1995 — APPROVED** [review]
This all LGTM. I left one comment. But if you think that my suggestions makes the code style worse, you can just discard it.

Feel free to r=me after rebase and potentially addressing my comment.

File: clippy_lints/src/methods/range_zip_with_len.rs (hunk 1 above)
Comment (Jarcho) [reply clippy_lints/src/methods/range_zip_with_len.rs]: Done.

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**